Error loading BeanIntrospection for type [class example.micronaut.TeamConfiguration]: Duplicate key

Using KSP with Serde (and the latest Micronaut Core) in the `micronaut-configuration` guide fails with

```
Caused by: io.micronaut.serde.exceptions.SerdeException: Error loading BeanIntrospection for type [class example.micronaut.TeamConfiguration]: Duplicate key
	at io.micronaut.serde.support.serializers.ObjectSerializer$4.tryToFindSerializer(ObjectSerializer.java:218)
	at io.micronaut.serde.support.serializers.ObjectSerializer$RuntimeTypeSerializer.lambda$getSerializer$0(ObjectSerializer.java:280)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1708)
	at io.micronaut.serde.support.serializers.ObjectSerializer$RuntimeTypeSerializer.getSerializer(ObjectSerializer.java:278)
	at io.micronaut.serde.support.serializers.ObjectSerializer$RuntimeTypeSerializer.serialize(ObjectSerializer.java:245)
	at io.micronaut.serde.jackson.JacksonJsonMapper.writeValue(JacksonJsonMapper.java:114)
	at io.micronaut.serde.jackson.JacksonJsonMapper.writeValue(JacksonJsonMapper.java:193)
	at io.micronaut.json.body.JsonMessageHandler.writeTo(JsonMessageHandler.java:134)
	... 60 common frames omitted
Caused by: io.micronaut.core.beans.exceptions.IntrospectionException: Error loading BeanIntrospection for type [class example.micronaut.TeamConfiguration]: Duplicate key
	at io.micronaut.core.beans.DefaultBeanIntrospector.findIntrospection(DefaultBeanIntrospector.java:104)
	at io.micronaut.serde.support.DefaultSerdeIntrospections.getSerializableIntrospection(DefaultSerdeIntrospections.java:77)
	at io.micronaut.serde.support.serializers.SerBean.<init>(SerBean.java:107)
	at io.micronaut.serde.support.serializers.ObjectSerializer.create(ObjectSerializer.java:202)
	at io.micronaut.serde.support.serializers.ObjectSerializer.lambda$getSerBean$0(ObjectSerializer.java:193)
	at io.micronaut.core.util.SupplierUtil$2.get(SupplierUtil.java:79)
	at io.micronaut.serde.support.serializers.ObjectSerializer.getSerBean(ObjectSerializer.java:194)
	at io.micronaut.serde.support.serializers.ObjectSerializer.createSpecificInternal(ObjectSerializer.java:93)
	at io.micronaut.serde.support.serializers.ObjectSerializer.createSpecific(ObjectSerializer.java:86)
	at io.micronaut.serde.jackson.JacksonJsonMapper.writeValue(JacksonJsonMapper.java:112)
	... 62 common frames omitted
Caused by: java.lang.IllegalArgumentException: Duplicate key
	at io.micronaut.core.util.StringIntMap.put(StringIntMap.java:74)
	at io.micronaut.inject.beans.AbstractInitializableBeanIntrospection.<init>(AbstractInitializableBeanIntrospection.java:97)
	at example.micronaut.$TeamConfiguration$Introspection.<init>(Unknown Source)
	at example.micronaut.$TeamConfiguration$IntrospectionRef.load(Unknown Source)
	at io.micronaut.core.beans.DefaultBeanIntrospector.lambda$findIntrospection$0(DefaultBeanIntrospector.java:95)
	at java.base/java.util.Optional.map(Optional.java:260)
	at io.micronaut.core.beans.DefaultBeanIntrospector.findIntrospection(DefaultBeanIntrospector.java:91)
	... 71 common frames omitted
```

### Steps to reproduce:

- Open the micronaut-guides module
- Check out the `sdelamo-patch-3` for micronaut-4
- Remove `"kotlinFeatures": ["kapt"]` from `guides/micronaut-configuration/metadata.json`
- Run `./gradlew clean micronautConfigurationBuild --no-build-cache`

### Expected behavior

All the tests should pass

### Actual behavior

The Kotlin/Gradle test fails (where we use KSP)

--- a/inject-kotlin/src/main/kotlin/io/micronaut/kotlin/processing/visitor/KotlinClassElement.kt
+++ b/inject-kotlin/src/main/kotlin/io/micronaut/kotlin/processing/visitor/KotlinClassElement.kt
@@ -372,17 +372,20 @@ internal open class KotlinClassElement(
             }
 
         val allProperties: MutableList<PropertyElement> = mutableListOf()
+        allProperties.addAll(enclosedElementsQuery.getEnclosedElements(this, eq))
         // unfortunate hack since these are not excluded?
         if (hasDeclaredStereotype(ConfigurationReader::class.java)) {
             val configurationBuilderQuery = ElementQuery.of(PropertyElement::class.java)
                 .annotated { it.hasDeclaredAnnotation(ConfigurationBuilder::class.java) }
                 .onlyInstance()
-            val configBuilderProps =
-                enclosedElementsQuery.getEnclosedElements(this, configurationBuilderQuery)
-            allProperties.addAll(configBuilderProps)
+                .onlyAccessible(this)
+            enclosedElementsQuery.getEnclosedElements(this, configurationBuilderQuery)
+                .forEach { e ->
+                    if (!allProperties.contains(e)) {
+                        allProperties.add(e)
+                    }
+                }
         }
-
-        allProperties.addAll(enclosedElementsQuery.getEnclosedElements(this, eq))
         val propertyNames = allProperties.map { it.name }.toSet()
 
         val resolvedProperties: MutableList<PropertyElement> = mutableListOf()

--- a/inject-kotlin/src/test/groovy/io/micronaut/kotlin/processing/beans/configproperties/ConfigurationPropertiesSpec.groovy
+++ b/inject-kotlin/src/test/groovy/io/micronaut/kotlin/processing/beans/configproperties/ConfigurationPropertiesSpec.groovy
@@ -18,6 +18,7 @@ package io.micronaut.kotlin.processing.beans.configproperties
 import io.micronaut.context.ApplicationContext
 import io.micronaut.context.DefaultApplicationContext
 import io.micronaut.context.env.PropertySource
+import io.micronaut.core.beans.BeanIntrospection
 import io.micronaut.core.util.CollectionUtils
 import spock.lang.Specification
 
@@ -144,5 +145,41 @@ class ConfigurationPropertiesSpec extends Specification {
         cleanup:
             context1.close()
             context2.close()
+    }
+
+    void "test team configuration admin configuration builder "() {
+        given:
+        ApplicationContext ctx = ApplicationContext.run(ApplicationContext, [
+                "team.name": 'evolution',
+                "team.color": 'green',
+                "team.player-names": ['Nirav Assar', 'Lionel Messi'],
+                "team.team-admin.manager": "Jerry Jones", // <1>
+                "team.team-admin.coach": "Tommy O'Neill",
+                "team.team-admin.president": "Mark Scanell"
+        ])
+
+        when:
+        TeamConfiguration teamConfiguration = ctx.getBean(TeamConfiguration)
+        TeamAdmin teamAdmin = teamConfiguration.builder.build() // <2>
+
+        then:
+        BeanIntrospection.getIntrospection(TeamConfiguration) != null
+        teamConfiguration.name == "evolution"
+        teamConfiguration.color == "green"
+        teamConfiguration.playerNames[0] == "Nirav Assar"
+        teamConfiguration.playerNames[1] == "Lionel Messi"
+
+        // check the builder has values set
+        teamConfiguration.builder.manager == "Jerry Jones"
+        teamConfiguration.builder.coach == "Tommy O'Neill"
+        teamConfiguration.builder.president == "Mark Scanell"
+
+        // check the object can be built
+        teamAdmin.manager == "Jerry Jones" // <3>
+        teamAdmin.coach == "Tommy O'Neill"
+        teamAdmin.president == "Mark Scanell"
+
+        cleanup:
+        ctx.close()
     }
 }

--- a/inject-kotlin/src/test/kotlin/io/micronaut/kotlin/processing/beans/configproperties/TeamConfiguration.kt
+++ b/inject-kotlin/src/test/kotlin/io/micronaut/kotlin/processing/beans/configproperties/TeamConfiguration.kt
@@ -1,0 +1,30 @@
+package io.micronaut.kotlin.processing.beans.configproperties
+
+import io.micronaut.context.annotation.ConfigurationBuilder
+import io.micronaut.context.annotation.ConfigurationProperties
+import io.micronaut.core.annotation.Introspected
+
+@ConfigurationProperties("team")
+@Introspected
+class TeamConfiguration  {
+    var name: String? = null
+    var color: String? = null
+    var playerNames: List<String>? = null
+    @ConfigurationBuilder(prefixes = ["with"], configurationPrefix = "team-admin") // <3>
+    var builder = TeamAdmin.Builder() // <4>
+}
+
+data class TeamAdmin(
+    val manager: String?,
+    val coach: String?,
+    val president: String?) { // <1>
+    data class Builder( // <2>
+        var manager: String? = null,
+        var coach: String? = null,
+        var president: String? = null) {
+        fun withManager(manager: String) = apply { this.manager = manager } // <3>
+        fun withCoach(coach: String) = apply { this.coach = coach }
+        fun withPresident(president: String) = apply { this.president = president }
+        fun build() = TeamAdmin(manager, coach, president) // <4>
+    }
+}

